roachtest: add rolling restart test

pav-kv opened the pull request:
Epic: none
Release note: none

pav-kv committed:
roachtest: add rolling restart test

Epic: none
Release note: none

pav-kv committed:
fixup

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/135153)
<!-- Reviewable:end -->

tbg committed:
more ranges, restart into new binary

Have `cockroach-old` and `cockroach-new` in place[^1] and run:

```
bin/roachtest run restart/rolling-and-rolling --artifacts "repro/$RANDOM" --cockroach cockroach-old --count 400
```

[^1]: https://cockroachlabs.slack.com/archives/C07VDN3CA3U/p1731572338960789?thread_ts=1731546467.192059&cid=C07VDN3CA3U

tbg committed:
fix bug

stevendanna commented on pkg/cmd/roachtest/tests/restart_rolling_and_rolling.go:
```diff
@@ -0,0 +1,115 @@
+	restart := func(transitionToNew bool) {
+		for _, n := range c.CRDBNodes() {
+			m.ExpectDeath()
+			t.Status(fmt.Sprintf("Stopping node %d.", n))
+			c.Stop(ctx, t.L(), option.DefaultStopOpts(), c.Node(n))
```
As a heads up, I this does a SIGKILL.  If we also want to test a mix of graceful shutdowns, we could do something like we do in some other tests:

```
	shouldUseSigKill := cfg.rng.Float64() > 0.5
	if shouldUseSigKill {
		t.L().Printf(`stopping node (using SIGKILL) %s`, target)
		if err := c.StopE(ctx, t.L(), option.DefaultStopOpts(), target); err != nil {
			return errors.Wrapf(err, "could not stop node %s", target)
		}
	} else {
		t.L().Printf(`stopping node gracefully %s`, target)
		if err := c.StopE(
			ctx, t.L(), option.NewStopOpts(option.Graceful(shutdownGracePeriod)), c.Node(cfg.shutdownNode),
		); err != nil {
			return errors.Wrapf(err, "could not stop node %s", target)
		}
	}
```

tbg committed:
alternate 10 min of high and low loads

tbg committed:
ignore workload error

tbg committed:
only 7.5k splits

tbg committed:
shorten light workload period

pav-kv committed:
use multi-store with WAL failover

pav-kv committed:
make the disk limiter multi-store

pav-kv committed:
let the workload run a bit before restarting
